pwndbg/commands/cymbol.py

```diff
@@ -0,0 +1,273 @@
+    editor_preference = os.getenv("EDITOR")
+    if not editor_preference:
+        editor_preference = os.getenv("VISUAL")
+    if not editor_preference:
+        editor_preference = "vi"
+
+    if cymbol_editor != "":
+        editor_preference = cymbol_editor
+
+    try:
+        subprocess.run([editor_preference, custom_structure_path], capture_output=False, check=True)
+    except Exception as exepction:
+        print(message.error("An error occured during opening the source file."))
+        print(message.error(f"Path to the custom structure: {custom_structure_path}"))
+        print(message.error("Please try to manually edit the structure."))
+        print(
+            message.error(
+                '\nTry to set a path to an editor with:\n\tset "cymbol-editor" /usr/bin/nano'
+            )
+        )
+        return
+
+    input(message.notice("Press enter to continue."))
+
+    load_custom_structure(custom_structure_name)
+
+
+@OnlyWhenStructureExists
+def remove_custom_structure(custom_structure_name, custom_structure_path):
+    os.remove(custom_structure_path)
+    print(message.success("Symbols are removed!"))
+    print(
+        message.notice(
+            "If the symbols are already loaded, please restart pwndbg to unload them from memory."
+        )
+    )
+
+
+@OnlyWhenStructureExists
+def load_custom_structure(custom_structure_name, custom_structure_path):
+    pwndbg_debug_symbols_output_file = generate_debug_symbols(custom_structure_path)
+    if not pwndbg_debug_symbols_output_file:
+        return
+    gdb.execute(f"add-symbol-file {pwndbg_debug_symbols_output_file}", to_string=True)
+    print(message.success("Symbols are loaded!"))
+
+
+@OnlyWhenStructureExists
+def show_custom_structure(custom_structure_name, custom_structure_path):
+    print()
+    with open(custom_structure_path, "r") as f:
+        print(f.read())
+    print()
+
+
+parser = argparse.ArgumentParser(
+    description="Add, show, load, edit, or delete custom structures in plain C"
+)
+parser.add_argument(
+    "-a",
+    "--add",
+    metavar="name",
+    help="Add a new custom structure",
+    default=None,
+    type=str,
+)
+parser.add_argument(
+    "-r",
+    "--remove",
+    metavar="name",
+    help="Remove an existing custom structure",
+    default=None,
+    type=str,
+)
+parser.add_argument(
+    "-e",
+    "--edit",
+    metavar="name",
+    help="Edit an existing custom structure",
+    default=None,
+    type=str,
+)
+parser.add_argument(
+    "-l",
+    "--load",
+    metavar="name",
+    help="Load an existing custom structure",
+    default=None,
+    type=str,
+)
+parser.add_argument(
+    "-s",
+    "--show",
+    metavar="name",
+    help="Show the source code of an existing custom structure",
+    default=None,
+    type=str,
+)
+
+
+@pwndbg.commands.ArgparsedCommand(parser)
+@pwndbg.commands.OnlyAmd64
+@pwndbg.commands.OnlyWhenRunning
+def cymbol(add, remove, edit, load, show):
+    if add:
```

Comment:
You can rewrite this as an if/elif/else statement.

Reply:
My latest commit should have fixed that.